mackerel-plugin-uwsgi-vassal/lib/uwsgi_vassal.go

```diff
@@ -0,0 +1,211 @@
+//     "respawn_count": 1,
+//     "tx": 0,
+//     "avg_rt": 0,
+//     "apps": [{
+//       "id": 0,
+//       "modifier1": 0,
+//       "mountpoint": "",
+//       "requests": 0,
+//       "exceptions": 0,
+//       "chdir": ""
+//     }]
+//   }, {
+//     "id": 2,
+//     "pid": 31760,
+//     "requests": 0,
+//     "exceptions": 0,
+//     "status": "idle",
+//     "rss": 0,
+//     "vsz": 0,
+//     "running_time": 0,
+//     "last_spawn": 1317235041,
+//     "respawn_count": 1,
+//     "tx": 0,
+//     "avg_rt": 0,
+//     "apps": [{
+//       "id": 0,
+//       "modifier1": 0,
+//       "mountpoint": "",
+//       "requests": 0,
+//       "exceptions": 0,
+//       "chdir": ""
+//     }]
+//   }, {
+//     "id": 3,
+//     "pid": 31761,
+//     "requests": 0,
+//     "exceptions": 0,
+//     "status": "idle",
+//     "rss": 0,
+//     "vsz": 0,
+//     "running_time": 0,
+//     "last_spawn": 1317235041,
+//     "respawn_count": 1,
+//     "tx": 0,
+//     "avg_rt": 0,
+//     "apps": [{
+//       "id": 0,
+//       "modifier1": 0,
+//       "mountpoint": "",
+//       "requests": 0,
+//       "exceptions": 0,
+//       "chdir": ""
+//     }]
+//   }, {
+//     "id": 4,
+//     "pid": 31762,
+//     "requests": 0,
+//     "exceptions": 0,
+//     "status": "idle",
+//     "rss": 0,
+//     "vsz": 0,
+//     "running_time": 0,
+//     "last_spawn": 1317235041,
+//     "respawn_count": 1,
+//     "tx": 0,
+//     "avg_rt": 0,
+//     "apps": [{
+//       "id": 0,
+//       "modifier1": 0,
+//       "mountpoint": "",
+//       "requests": 0,
+//       "exceptions": 0,
+//       "chdir": ""
+//     }]
+//   }
+// }
+
+// field types vary between versions
+
+// UWSGIWorker struct
+type UWSGIWorker struct {
+	Requests uint64 `json:"requests"`
+	Status   string `json:"status"`
+}
+
+// UWSGIWorkers sturct for json struct
+type UWSGIWorkers struct {
+	Workers []UWSGIWorker `json:"workers"`
+}
+
+// FetchMetrics interface for mackerelplugin
+func (p UWSGIVassalPlugin) FetchMetrics() (map[string]interface{}, error) {
+	stat := make(map[string]interface{})
+	stat["busy"] = uint64(0)
+	stat["idle"] = uint64(0)
+	stat["cheap"] = uint64(0)
+	stat["pause"] = uint64(0)
+	stat["requests"] = uint64(0)
+
+	var decoder *json.Decoder
+	if strings.HasPrefix(p.Socket, "unix://") {
+		conn, err := net.Dial("unix", strings.TrimPrefix(p.Socket, "unix://"))
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+		decoder = json.NewDecoder(conn)
+	} else if strings.HasPrefix(p.Socket, "http://") {
+		resp, err := http.Get(p.Socket)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+		decoder = json.NewDecoder(resp.Body)
+	}
+
+	var workers UWSGIWorkers
+	if err := decoder.Decode(&workers); err != nil {
+		return nil, err
+	}
+
+	for _, worker := range workers.Workers {
+		switch worker.Status {
+		case "idle", "busy", "cheap", "pause":
+			stat[worker.Status] = reflect.ValueOf(stat[worker.Status]).Uint() + 1
```

Comment:
Hmm……
Type converting using `reflect` seems difficult.
[go-mackerel-plugin](https://github.com/mackerelio/go-mackerel-plugin) simply uses `map[string]float64` as stats, you can switch from go-mackerel-plugin-helper to it.
(Or just using intermediate variable may help?)
```golang
intStats := make(map[string]int64)
for _, worker := range workers.Workers {
  intStats[worker.Status] = intStats[worker.Status] + 1
  intStats["requests"] = insStats["requests"] + 1
}
for k, v := range intStats {
  stats[k] = v
}
```